Plutarch/Api/V1/Value.hs

```diff
@@ -5,44 +5,63 @@ module Plutarch.Api.V1.Value (
   PValue (PValue),
   PCurrencySymbol (PCurrencySymbol),
   PTokenName (PTokenName),
+  PAssetClass (PAssetClass),
 ) where
 
-import qualified Plutus.V1.Ledger.Api as Plutus
+import qualified Plutus.V1.Ledger.Value as PlutusValue
 import qualified PlutusTx.Builtins.Internal as PT
 
 import Plutarch.Api.V1.AssocMap (PMap)
 import Plutarch.Lift (
   DerivePConstantViaNewtype (DerivePConstantViaNewtype),
+  PConstantRepr,
+  PConstanted,
   PLifted,
   PUnsafeLiftDecl,
+  pconstantFromRepr,
+  pconstantToRepr,
  )
 import Plutarch.Prelude
+import qualified PlutusTx
 
 newtype PTokenName (s :: S) = PTokenName (Term s PByteString)
   deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PTokenName PByteString)
 
-instance PUnsafeLiftDecl PTokenName where type PLifted PTokenName = Plutus.TokenName
+instance PUnsafeLiftDecl PTokenName where type PLifted PTokenName = PlutusValue.TokenName
 deriving via
-  (DerivePConstantViaNewtype Plutus.TokenName PTokenName PByteString)
+  (DerivePConstantViaNewtype PlutusValue.TokenName PTokenName PByteString)
   instance
-    (PConstant Plutus.TokenName)
+    (PConstant PlutusValue.TokenName)
 
 newtype PCurrencySymbol (s :: S) = PCurrencySymbol (Term s PByteString)
   deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PCurrencySymbol PByteString)
 
-instance PUnsafeLiftDecl PCurrencySymbol where type PLifted PCurrencySymbol = Plutus.CurrencySymbol
+instance PUnsafeLiftDecl PCurrencySymbol where type PLifted PCurrencySymbol = PlutusValue.CurrencySymbol
 deriving via
-  (DerivePConstantViaNewtype Plutus.CurrencySymbol PCurrencySymbol PByteString)
+  (DerivePConstantViaNewtype PlutusValue.CurrencySymbol PCurrencySymbol PByteString)
   instance
-    (PConstant Plutus.CurrencySymbol)
+    (PConstant PlutusValue.CurrencySymbol)
 
 newtype PValue (s :: S) = PValue (Term s (PMap PCurrencySymbol (PMap PTokenName PInteger)))
   deriving
     (PlutusType, PIsData)
     via (DerivePNewtype PValue (PMap PCurrencySymbol (PMap PTokenName PInteger)))
 
-instance PUnsafeLiftDecl PValue where type PLifted PValue = Plutus.Value
+instance PUnsafeLiftDecl PValue where type PLifted PValue = PlutusValue.Value
 deriving via
-  (DerivePConstantViaNewtype Plutus.Value PValue (PMap PCurrencySymbol (PMap PTokenName PInteger)))
+  (DerivePConstantViaNewtype PlutusValue.Value PValue (PMap PCurrencySymbol (PMap PTokenName PInteger)))
   instance
-    (PConstant Plutus.Value)
+    (PConstant PlutusValue.Value)
+
+newtype PAssetClass (s :: S) = PAssetClass (Term s (PBuiltinPair (PAsData PCurrencySymbol) (PAsData PTokenName)))
```

Comment:
I'm still debating if this should be a newtype to a `PBuiltinPair` or a `PTuple`